internal/resources/tanzupackageinstall/package_install_resource.go

```diff
@@ -187,8 +192,14 @@ func resourcePackageInstallInPlaceUpdate(ctx context.Context, d *schema.Resource
 		updateAvailable = true
 	}
 
-	if updateCheckForSpec(d, installSpec) {
-		err := CheckForUpdatedPackage(config, scopedFullnameData, installSpec)
+	specCheck, err := updateCheckForSpec(d, installSpec)
+	if err != nil {
+		log.Println("[ERROR] Unable to check spec has been updated.")
+		diag.FromErr(err)
+	}
```

Comment:
is this an ignorable error?

Reply:
No its not, so it is being returned in the diagnostics for the same reason.
`diag.FromErr(err)`

Reply:
please consider adding `return` statement, `return diag.FromErr(err)`

Reply:
Updated.